akka-actor-typed/src/main/scala/akka/actor/typed/internal/BehaviorImpl.scala

```diff
@@ -64,17 +64,25 @@ import akka.actor.typed.scaladsl.{ ActorContext ⇒ SAC }
   def intercept[O, I](interceptor: BehaviorInterceptor[O, I])(behavior: Behavior[I]): Behavior[O] =
     InterceptorImpl(interceptor, behavior)
 
-  class OrElseBehavior[T](first: Behavior[T], second: Behavior[T]) extends ExtensibleBehavior[T] {
-    override def receive(ctx: AC[T], msg: T): Behavior[T] = {
-      Behavior.interpretMessage(first, ctx, msg) match {
-        case _: UnhandledBehavior.type ⇒ Behavior.interpretMessage(second, ctx, msg)
-        case handled                   ⇒ handled
+  class OrElseInterceptor[T](next: Behavior[T]) extends BehaviorInterceptor[T, T] {
```

Comment:
it's not implementing `isSame` so how does this work when returning new `orElse` behaviors and such?
see `InterceptSpec` and issue https://github.com/akka/akka/issues/25504

Reply:
Ah indeed and there's no logical way to know it is the same Or and there's no real wrapping going on. I'll revert this to be an extensible behaviour and start it in a deferred.